crates/lib/src/qpu/quilc/mod.rs

```diff
@@ -50,6 +50,19 @@ pub(crate) fn compile_program(
     }
 }
 
+pub(crate) fn get_version_info(client: &Qcs) -> Result<String, Error> {
+    let config = client.get_config();
+    let endpoint = config.quilc_url();
+    let binding: HashMap<&str, &str> = HashMap::new();
+    let request = rpcq::RPCRequest::new("get_version_info", &binding);
```

Comment:
What is the best way to pass in an empty set of params to `rpcq::RPCRequest::new`? I don't love initializing this empty `HashMap` just for that purpose, but the function needs something, and the empty types I tried crash quilc. cc'ing: @Shadow53 for their Rust expertise.

Reply:
`HashMap::new` should be fine. The standard library generally tries to avoid allocating until necessary, which [is the case](https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.new) with `HashMap::new`.

Could also try passing in `()` if you haven't already. I think that implements `Serialize`. Only other option I can think to try is passing `None`, but I'm pretty sure the compiler would complain about not knowing what type the `Option` should contain.

Reply:
I tried `()` first, which also unfortunately doesn't work. Rust doesn't mind, but `quilc` throws a pretty gnarly error, dies, and leaves the Rust library hanging.

```
Unhandled TYPE-ERROR in thread #<SB-THREAD:THREAD "RPC-server-thread-0" RUNNING
                                  {1005A834B3}>:
  The value
    NIL
  is not of type
    HASH-TABLE
```

Reply:
Sounds good. Good to know that it requires a hash map.

It's strictly unrelated to this change, but it would be a good idea to update the documentation and/or signature for making a new `RPCRequest` to only allow hash maps, e.g. using (pseudo code) `binding: HashMap<K=String, V=String>` rather than `binding: T = HashMap<String, String>`

Reply:
I like the idea of hardening the type that gets accepted, but I think only allowing HashMap would be too restrictive. For example, the compile function uses the `QuilcParams` struct. We only use a HashMap here as a quick way of creating an empty set of params.